src/hotspot/share/gc/parallel/psOldGen.hpp

```diff
@@ -158,9 +134,10 @@ class PSOldGen : public CHeapObj<mtGC> {
   // Calculating new sizes
   void resize(size_t desired_free_space);
 
-  // Allocation. We report all successful allocations to the size policy
-  // Note that the perm gen does not use this method, and should not!
-  HeapWord* allocate(size_t word_size);
+  HeapWord* allocate(size_t word_size) {
```

Comment:
Before this change there has been a small semantical difference between `allocate()` and `par_allocate()`. The former also updated the size policy, which seem to be missing now in `ParallelScavengeHeap::mem_allocate_old_gen()` and `ParallelScavengeHeap::failed_mem_allocate()`.

Reply:
Oops, I forgot about that and got overly enthusiastic about code deletion.  Thanks for spotting it.